libs/counterfactuals/src/lib/CounterfactualList.tsx

```diff
@@ -163,8 +163,13 @@ export class CounterfactualList extends React.Component<
     const data = _.cloneDeep(items[idx]);
     Object.keys(data).forEach((k) => {
       data[k] = data[k] === "-" ? items[0][k] : data[k];
-      const keyIndex =
-        this.props.data?.feature_names_including_target.indexOf(k);
+      const metaDict = this.context.jointDataset.metaDict;
+      const columnKey = Object.keys(metaDict).find(
```

Comment:
nit: you can move some of these outside of the foreach for clarity and performance, specifically:
```
const metaDict = this.context.jointDataset.metaDict;
const metaDictKeys = Object.keys(metaDict)
 Object.keys(data).forEach((k) => {
      data[k] = data[k] === "-" ? items[0][k] : data[k];
      const columnKey = metaDictKeys.find(
```

Reply:
Updated!